### ISSUE TITLE
How to use properly search with hasMany, I cant get work

### ISSUE THREAD

**Issue by diorges** [yajra/laravel-datatables#625]
### Summary of problem or feature request

I`m trying datatables with hasMany but I cant get working, the column CONTATO get null. I want to search this column only, because Assinante (customer) hasMany Contato.

The error : DataTables warning: table id=assinantesLista - Requested unknown parameter 'contatos.nome' for row 0.

### Code snippet of problem

```php
<table id="assinantesLista" class="table table-bordered table-striped">
                        <thead>
                            <tr>
                                <th>ID</th>
                                <th>NOME</th>
                                <th>CONTATO</th>
                                <th>FONE</th>
                                <th>TIPO</th>
                                <th></th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <TD colspan="6"></TD>
                            </tr>
                        </tbody>
                    </table>

var assinantesLista = $("#assinantesLista").DataTable({
                processing: true,
                serverSide: true,
                ajax: '{{route('admin.assinantes.json')}}',
                "aoColumnDefs": [
                    {"visible": false, "targets": 2}
                ],
                columns: [
                    {data: 'id', name: 'assinantes.id'},
                    {data: 'nome', name: 'assinantes.nome'},
                    {data: 'contatos.nome', name: 'contatos.nome'},
                    {data: 'fone', name: 'assinantes.fone'},
                    {data: 'tipo', name: 'assinantes.tipo'},
                    {data: 'actions', name: 'actions', orderable: false, searchable: false},
                ]
            });

public function datatables(Request $request){
        $assinantes = $this->assinante->with('contatos');
        return Datatables::of($assinantes)
            ->editColumn('nome', function($assinante) {
                return $assinante->tipo=="J"?$assinante->nome:$assinante->contatoPrincipal()->nome.' '.$assinante->contatoPrincipal()->snome;
            })
            ->editColumn('fone', function($assinante) {
                return $assinante->tipo=="J"?$assinante->fone:$assinante->contatoPrincipal()->fone;
            })
            ->editColumn('tipo', '{{$tipo=="J"?"Empresarial":"Residencial"}}')
            ->addColumn('actions', '<div class="btn-group" role="group" aria-label="Ações"><a href="{{route(\'admin.assinantes.view\', [\'id\' => $id])}}" type="button" class="btn btn-default btn-xs"><i class="fa fa-search"></i></a>')
            ->make(true);
    }
```

### System details

- Operating System Ubuntu Server 14.4 
- PHP Version 5.5.9
- Laravel Version 5.2
- Laravel-Datatables Version 6.14.0

**Comment by yajra** [comment]
@diorges hasMany will return an array which means that you cannot access `contatos.nome` from response. Maybe you can do something like below:

```php
->addColumn('contatos_nome', function($assinante) {
    $contatos = $assinante->contatos;
    // explode or concat your array
    return $contatos_string_version;
})
```

On your js:

```js
{data: 'contatos_nome', name: 'contatos.nome'},
```

**Comment by yajra** [comment]
@diorges can try again on v6.18.0, a bug was found and fixed that is related to model relations. Thanks!

**Comment by yajra** [comment]
@diorges another fix related to HasMany was released on v6.18.1. Your issue should be fixed on the latest version. I also added a demo for has many. See this for ref: https://datatables.yajrabox.com/relation/has-many
